File: files/CSEC 09.06.R
library(tidyverse)
library(readxl)

input = read_excel("files/Excel Challenge 9th June .xlsx", range = "C2:F13")
test  = read_excel("files/Excel Challenge 9th June .xlsx", range = "H2:H10") %>%
  arrange(Students)

result = input %>%
  pivot_longer(cols = -c(1)) %>%
  count(value) %>%
  filter(n == 1) %>%
  select(Students = value) %>%
  arrange(Students)

result
# Eric and Fred attended 2 trainings

File: files/2024-06-09/CSEC 09.06.py
import pandas as pd

input = pd.read_excel("files/Excel Challenge 9th June .xlsx", usecols="C:F", skiprows=1)
test = pd.read_excel("files/Excel Challenge 9th June .xlsx",  usecols="H:H", skiprows=1, nrows = 8) \
    .sort_values(by="Students").reset_index(drop=True)

result = input.melt(id_vars=["Start"], var_name="Subject", value_name="Students")
result = result.groupby("Students").count().reset_index()
result = result[result["Start"] == 1]["Students"].reset_index(drop=True)
print(result)
# Eric and Fred has 2 trainings